apis/python/src/tiledbsoma/tiledb_array.py

```diff
@@ -36,31 +36,36 @@ def schema(self) -> pa.Schema:
         """
         return get_arrow_schema_from_tiledb_uri(self.uri, self._ctx)
 
-    def _tiledb_open(self, mode: str = "r") -> tiledb.Array:
-        """
-        This is just a convenience wrapper allowing 'with self._tiledb_open() as A: ...' rather than 'with tiledb.open(self._uri) as A: ...'.
-        """
-        if mode not in ["r", "w"]:
-            raise ValueError(f'expected mode to be one of "r" or "w"; got "{mode}"')
-        # This works in either 'with self._tiledb_open() as A:' or 'A = self._tiledb_open(); ...; A.close().  The
-        # reason is that with-as invokes our return value's __enter__ on return from this method,
-        # and our return value's __exit__ on exit from the body of the with-block. The tiledb
-        # array object does both of those things. (And if it didn't, we'd get a runtime AttributeError
-        # on with-as, flagging the non-existence of the __enter__ or __exit__.)
-        return tiledb.open(self._uri, mode=mode, ctx=self._ctx)
+    # lazy tiledb.Array handle for reuse while self is "open"
+    _open_tiledb_array: Optional[tiledb.Array] = None
+
+    @property
+    def _tiledb_obj(self) -> tiledb.Array:
+        "get the open tiledb.Array handle (opening it if needed)"
```

Comment:
I"m curious why we use a lazy open of the underlying resource, rather than doing it at the time of the original `open()` call?  

The delayed call to `tiledb.open` means that errors associated with this will be delayed.  For example, if the `tiledb.open` fails (say due to bad URI), it will raise during a `read`, rather than during the `open`.  Raising early (during `open`) feels more natural (principle of lease surprise?) to me.

Reply:
I'd also add that lazy opening negates some of the performance optimizations we were looking for by making the 
"first" query slower than subsequent ones.

I think that as post-1.0 nearly all of this logic must be moved into C++ outside of python/R so I'm not going to push for one or the other here. Moving this to C++ will yield many more improvements for Reads and longer term writes.

Reply:
Resolving, as this is now eager.